modules/server/src/network/resolvers/index.ts

```diff
@@ -27,18 +32,25 @@ export const createResolvers = (configs: NodeConfig[]) => {
 		Query: {
 			network: async (
 				parent: NetworkSearchRoot,
-				args: {},
+				// type should match gql typedefs
+				args: NetworkAggregationArgs,
 				context: unknown,
 				info: GraphQLResolveInfo,
 			) => {
 				const requestedFieldsMap = resolveInfoToMap(info, 'aggregations');
 
+				try {
+					isSQONFilter(args.filters);
+				} catch (err) {
+					throw `SQON filters are not valid. ${JSON.stringify(err)}`;
```

Comment:
Does GQL want us to throw strings instead of an error like `new Error(message)`?

It seems weird that you created `isSQONFilter` to throw an error and then immediately caught and threw a different error. Would be more reusable to have the sqon type check function return a Result with success containing the SQON and the failure containing the error message you want to include in your new thrown error.

```ts
import SQONBuilder, {SQON} from '@overture-stack/sqon-builder';

/**
 * Attempts to convert a variable to a SQON. A Result is returned, where if successful
 * the type checked SQON object is returned, and if a failure occurs then the error data
 * from the SQONBuilder is returned as the failure message.
 * 
 * @param filter 
 * @returns 
 */
export const convertToSqon = (filter: unknown): Result<SQON, 'INVALID_SQON', void> => {
	try {
		const output = SQONBuilder.from(filter);
		return success(output.toValue());
	} catch (error: unknown) { 
		let message = `${error}`; // Will convert the error to a string, works fine with the expected ZodError without reuiring importing Zod to arranger
		return failure('INVALID_SQON', message)
	}
};
```

With this, you can also update the `NetworkAggregationArgs` to have `filters` as a `SQON` instead of `object`.

Reply:
Bonus, I'd rename your file `httpResponse.ts` to be `result.ts`.